infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -339,6 +343,52 @@ public synchronized void handleFailureDetectedMsg(CorfuPayloadMsg<FailureDetecto
     }
 
     /**
+     * Adds a new node to the layout.
+     *
+     * @param msg corfu message containing ADD_NODE_REQUEST
+     * @param ctx netty ChannelHandlerContext
+     * @param r   server router
+     */
+    @ServerHandler(type = CorfuMsgType.ADD_NODE_REQUEST, opTimer = metricsPrefix + "add-node")
+    public void handleAddNodeRequest(CorfuPayloadMsg<AddNodeRequest> msg,
+                                     ChannelHandlerContext ctx, IServerRouter r,
+                                     boolean isMetricsEnabled) {
+        if (isShutdown()) {
+            log.warn("Management Server received {} but is shutdown.", msg.getMsgType().toString());
+            r.sendResponse(ctx, msg, new CorfuMsg(CorfuMsgType.NACK));
+            return;
+        }
+        // This server has not been bootstrapped yet, ignore all requests.
+        if (!checkBootstrap(msg, ctx, r)) {
```

Comment:
should this be moved to a lower layer? Also, why is the check responding instead of the main handler ?

Reply:
Will move this check